client/state/purchases/selectors.js

```diff
@@ -50,6 +52,18 @@ export const getSitePurchases = ( state, siteId ) => (
 	getPurchases( state ).filter( purchase => purchase.siteId === siteId )
 );
 
+export const getIncludedDomainPurchase = ( state, purchase ) => {
+	if ( ! purchase || ! isSubscription( purchase ) ) {
+		return null;
+	}
+
+	const { includedDomain } = purchase;
+	const sitePurchases = getSitePurchases( state, purchase.siteId );
+	const domainPurchase = sitePurchases.find( p => ( isDomainMapping( p ) || isDomainRegistration( p ) ) && includedDomain === p.meta );
```

Comment:
We usually try to use full words for variables. The line will be a bit long in that case though so we should probably format that code on multiple lines.